src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -329,10 +346,48 @@ internal static void ApplyPolymorphismOptions(this JsonObject schema, JsonSchema
     /// <summary>
     /// Set the x-schema-id property on the schema to the identifier associated with the type.
     /// </summary>
-    /// <param name="schema">The <see cref="JsonObject"/> produced by the underlying schema generator.</param>
-    /// <param name="context">The <see cref="JsonSchemaGenerationContext"/> associated with the current type.</param>
-    internal static void ApplySchemaReferenceId(this JsonObject schema, JsonSchemaGenerationContext context)
+    /// <param name="schema">The <see cref="JsonNode"/> produced by the underlying schema generator.</param>
+    /// <param name="context">The <see cref="JsonSchemaExporterContext"/> associated with the current type.</param>
+    internal static void ApplySchemaReferenceId(this JsonNode schema, JsonSchemaExporterContext context)
     {
         schema[OpenApiConstants.SchemaId] = context.TypeInfo.GetSchemaReferenceId();
     }
+
+    /// <summary>
+    /// Support applying nullability status for reference types provided as a parameter.
+    /// </summary>
+    /// <param name="schema">The <see cref="JsonNode"/> produced by the underlying schema generator.</param>
+    /// <param name="parameterInfo">The <see cref="ParameterInfo" /> associated with the schema.</param>
+    internal static void ApplyNullabilityContextInfo(this JsonNode schema, ParameterInfo parameterInfo)
+    {
+        if (parameterInfo.ParameterType.IsValueType)
+        {
+            return;
+        }
+
+        var nullabilityInfo = _nullabilityInfoContext.Create(parameterInfo);
```

Comment:
Nullability information is stored in the `JsonPropertyInfo`/`JsonParameterInfo` types, so you can avoid taking a dependency on a reflection-based components that might be producing inconsistent results in trimmed applications.

Reply:
> Nullability information is stored in the `JsonPropertyInfo`/`JsonParameterInfo` types, so you can avoid taking a dependency on a reflection-based component that might be producing inconsistent results in trimmed applications.

Are you referring to `isGetNullable`/`IsSetNullable` here?

Reply:
Correct

Reply:
Got it!

How do we resolve JsonParameterInfo? We're no longer passing ParameterInfo to the exporter API. Is there an explicit API for resolving a JsonParameterInfo instance?

Reply:
Yep, it's now a property on JsonPropertyInfo itself.

Reply:
> Yep, it's now a property on JsonPropertyInfo itself.

Ah, I dug into this. It looks like this applies specifically for constructor parameters. For our particular case, we care about parameters on a method.